scipy/stats/_morestats.py

```diff
@@ -4386,6 +4387,12 @@ def directional_stats(samples, *, axis=0, normalize=True):
         raise ValueError("samples must at least be two-dimensional. "
                          f"Instead samples has shape: {tuple(samples.shape)}")
     samples = xp.moveaxis(samples, axis, 0)
+
+    if is_marray(xp):
+        _xp = array_namespace(samples.mask)
+        mask = _xp.any(samples.mask, axis=-1, keepdims=True)
+        samples = xp.asarray(samples.data, mask=mask)
```

Comment:
@dschmitz89 Do you have a sense for what the preferred behavior would be when, say, only one element of a row is masked? 

The behavior here propagates the mask to all elements of the row (observation). This matches the behavior of ignoring the entire observation.

It's unclear whether another option is available. Suppose there is only one masked element in the entire array. If we ignore *just* the originally masked element, I think this would have have exactly the same affect on the other components as if the masked element were an unmasked zero. (The effect is a little different on the component that has a masked entry.) That doesn't seem quite right.

Reply:
I don't have a strong opinion but the implemented policy would work for me as a user.